xarray/backends/file_manager.py

```diff
@@ -355,13 +355,15 @@ def __init__(
         opener: Callable[..., T_File],
         *args: Any,
         mode: Any = _OMIT_MODE,
+        lock: Lock | None | Literal[False] = None,
         kwargs: Mapping[str, Any] | None = None,
     ):
         kwargs = {} if kwargs is None else dict(kwargs)
         self._opener = opener
         self._args = args
         self._mode = "a" if mode == "w" else mode
         self._kwargs = kwargs
+        self._lock = lock
```

Comment:
The rest of PickleableFileManager needs updates to use `lock`, at least in the `close` and `__getstate__`/`__setstate__` methods.